File: leetcode/robot_return.py
def judge_circle(moves: str) -> bool:
    # return moves.count('U') == moves.count('D') and moves.count('L') == moves.count('R')
    x = y = 0
    for move in moves:
        if move == 'U':
            y += 1
        elif move == 'D':
            y -= 1
        elif move == 'L':
            x -= 1
        elif move == 'R':
            x += 1
    return x == y == 0


print(judge_circle('UDLR'))

File: leetcode/robotReturn.js
// const judgeCircle = moves => [...moves.matchAll(/(U)/g)].length === [...moves.matchAll(/(D)/g)].length && [...moves.matchAll(/(L)/g)].length === [...moves.matchAll(/(R)/g)].length;

const judgeCircle = moves => {
    let [x, y] = [0, 0]
    for (let move of moves) {
        if (move === 'U') y++;
        else if (move === 'D') y--;
        else if (move === 'L') x--;
        else if (move === 'R') x++;
    }
    return x === 0 && y === 0;
}


console.log(judgeCircle('UDLR'))